src/modules/wishlist/api/wishlist.api.ts

```diff
@@ -0,0 +1,36 @@
+import { useMutation } from '@tanstack/react-query';
+import { useApiClient } from '#/shared/hooks/api-client.hook';
+import { type BaseResponse } from '#/shared/types/base-response.type';
+import { type WishlistItem } from '../types/wishlist-item.type';
+import { type AxiosInstance } from 'axios';
+
+interface MutationData {
+  accommodationId: string;
+}
+
+interface MutationParams {
+  action: 'add' | 'remove';
+  data: MutationData;
+}
+
+export function useWishlistMutation() {
+  const { privateApiClient } = useApiClient();
```

Comment:
Why?

Reply:
> Why?

Because I want to make this request with Auth header? Should I do it differently

Reply:
> Why?

Fixed